control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs

```diff
@@ -266,3 +229,111 @@ async fn check_and_drain_node(
     }
     PollResult::Ok(PollerState::Idle)
 }
+
+async fn drain_volume_target(
+    context: &PollContext,
+    node_spec: &OperationGuardArc<NodeSpec>,
+    mut guarded_vol_spec: OperationGuardArc<VolumeSpec>,
+) -> Either<bool, VolumeId> {
+    let Some(target) = guarded_vol_spec.as_ref().target() else {
+        return Either::Left(false);
+    };
+
+    let drain_node_id = node_spec.as_ref().id();
+    let Some(target_node) =
+        drain_target_node(context, target.node(), drain_node_id, &guarded_vol_spec)
```

Comment:
(grumble) IMO I'd prefer it to be called get_drain_target_node() because at first glance the function name makes you think it is doing the drain.
But I suppose we have to stick to the rustacean way...

Reply:
I agree, reading it now sounds confusing.
I'll rename to select_drain_target_node.